src/toast_handler.rs

```diff
@@ -0,0 +1,114 @@
+use crate::views::hardware_button::hw_description;
+use crate::views::hardware_view::HardwareView;
+use crate::widgets::toast::{Status, Toast};
+use crate::Message;
+use iced::Command;
+
+#[derive(Debug, Clone)]
+pub enum ToastMessage {
+    VersionToast,
+    HardwareDetailsToast,
+    UnsavedChangesToast,
+    UnsavedChangesExitToast,
+    Close(usize),
+    Timeout(f64),
+}
+
+pub struct ToastHandler {
+    pub toasts: Vec<Toast>,
```

Comment:
when you can leave these private (without accessor methods to read them....then you will know you have encapsulated the logic well within the toast handler)